libraries/shared/watcher/storage_watcher_test.go

```diff
@@ -394,15 +414,25 @@ func SharedExecuteBehavior(input *ExecuteInput) {
 
 			It("marks diff as 'unrecognized' when transforming the diff returns a ErrKeyNotFound error", func() {
 				mockTransformer.ExecuteErr = types.ErrKeyNotFound
-				setGetDiffsErrors(storageWatcher.DiffStatus, mockDiffsRepository, []error{nil, types.ErrKeyNotFound})
+				setGetDiffsErrors(storageWatcher.DiffStatus, mockDiffsRepository, []error{nil, fakes.FakeError})
 
 				err := storageWatcher.Execute()
 
 				Expect(err).To(HaveOccurred())
-				Expect(err).To(MatchError(types.ErrKeyNotFound))
+				Expect(err).To(MatchError(fakes.FakeError))
 				Expect(mockDiffsRepository.MarkUnrecognizedPassedID).To(Equal(fakePersistedDiff.ID))
 			})
 
+			It("does not return ErrKeyNotFoundError if key unrecognized", func() {
```

Comment:
How is this different from the above test?

Reply:
sql.ErrNoRows vs ErrKeyNotFound - added parallel structure to hopefully clarify a bit